src/reconcile/rec_hs.c

```diff
@@ -347,14 +349,17 @@ __rec_hs_cursor_pos(WT_SESSION_IMPL *session, WT_CURSOR *hs_cursor, uint32_t btr
      * If we find a key with a timestamp larger than or equal to the specified timestamp then the
      * specified timestamp must be mixed mode.
      */
-    char ts_string[4][WT_TS_INT_STRING_SIZE];
+    char ts_string[6][WT_TS_INT_STRING_SIZE];
```

Comment:
I think we should move this assert to before WT_RET(ret); otherwise, we don't assert if we get a not found error.